tools/caffe.cpp

```diff
@@ -12,13 +12,16 @@ using caffe::Blob;
 using caffe::Caffe;
 using caffe::Net;
 using caffe::Layer;
+using caffe::Solver;
 using caffe::shared_ptr;
+using caffe::string;
 using caffe::Timer;
 using caffe::vector;
+using std::ostringstream;
 
-
-DEFINE_int32(gpu, -1,
-    "Run in GPU mode on given device ID.");
```

Comment:
Why not leave this flag? There's no sense in breaking backwards-compatibility unnecessarily, is there?

I vote for `-gpu 1` and `-gpus 1` being equivalent.